src/models/util/ParseCalChart31.ts

```diff
@@ -0,0 +1,355 @@
+/**
+ * CalChart3.1 was a bytestream with a series of 4 char codes describing the
+ * next section.  It was more complicated than CalChart3.4.
+ *
+ * Description of the CalChart file format layout, in modified Extended
+ * Backus–Naur Form
```

Comment:
I'm not familiar with this format, so maybe somebody familiar can look at this later to double check this... Maybe make an issue for Kevin? This is not a blocker for merging.